### DISCUSSION TITLE
Functions in python

### DISCUSSION THREAD

**Discussion by tofetpuzo** [tofetpuzo/Kaisen#33] [General]
Using functions in a list.

**Comment by tofetpuzo** [comment]
print(abs(items_people_bought))
          ^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: bad operand type for abs(): 'list'

**Comment by Shriharsh-Deshmukh** [comment]
The error occurs because `abs()` only works with numeric types, not lists. To fix it, ensure you're applying `abs()` to individual numeric elements within the list. Example:

```python
print([abs(item) for item in items_people_bought])
```

**Reply by tofetpuzo** [reply]
Thanks, I am aware was just trying to demonstrate somethings to some people I mentor thanks.

**Reply by tofetpuzo** [reply]
@Shriharsh-Deshmukh  are you by any chance looking to work on a collaborative open source project?

**Comment by TokunboM** [comment]
hello,so i wrote a function to call another but getting an error. Find attached the script
```
no_of_boys = 12
no_of_girls = 8
no_present = 15

def sum_of_pupils_in_class():
    print(f"{no_of_boys} + {no_of_girls}")
    print("All present!")

def sum_of_pupils_absents():
    print(f"{sum_of_pupils_in_class} - {no_present}")
    print("Are absent!")

```

sum_of_pupils_absents()

**Reply by tofetpuzo** [reply]
Hi Hi, what is the error can you copy and paste here?

**Reply by tofetpuzo** [reply]
```
no_of_boys = 12
no_of_girls = 8
no_present = 15


def sum_of_pupils_in_class():
    total_pupils = no_of_boys + no_of_girls
    return total_pupils


def sum_of_pupils_absents():
    total_absents = sum_of_pupils_in_class() - no_present
    return total_absents


sum_of_pupils_in_class()
print(sum_of_pupils_absents())
```
